ios/FluentUI/Core/Theme/FluentTheme.swift

```diff
@@ -6,19 +6,32 @@
 import SwiftUI
 
 /// Base class that allows for customization of global, alias, and control tokens.
-public class FluentTheme {
+@objc public class FluentTheme: NSObject {
     /// Initializes and returns a new `FluentTheme`, with optional custom global and alias tokens.
     ///
     /// - Parameters globalTokens: An optional customized instance of `GlobalTokens`.
     /// - Parameters aliasTokens: An optional customized instance of `AliasTokens`.
     ///
     /// - Returns: An initialized `FluentTheme` instance that leverages the aforementioned global and alias token overrides.
     public init(globalTokens: GlobalTokens? = nil, aliasTokens: AliasTokens? = nil) {
+        var hasCustomGlobalTokens = false
         if let globalTokens = globalTokens {
             self.globalTokens = globalTokens
+            hasCustomGlobalTokens = true
         }
+
+        // Alias tokens are trickier. We need a custom set any time we have custom
+        // global tokens (so that they can reference those custom tokens), even if
+        // the caller does not give us a custom set themselves.
         if let aliasTokens = aliasTokens {
             self.aliasTokens = aliasTokens
+        } else if hasCustomGlobalTokens {
+            self.aliasTokens = .init()
+        }
+
+        // If we have custom global tokens, our "custom" alias tokens from above must track them.
+        if hasCustomGlobalTokens {
+            self.aliasTokens.globalTokens = self.globalTokens
```

Comment:
Seems like we should be able to combine the two if hasCustomGlobalTokens. Or maybe move the alias tokens above the global tokens so we can set the aliasTokens.globalTokens in the  global tokens section?

Reply:
Yeah I agree, let me see how I can clean this up a bit.